pkg/apis/core/validation/shoot.go

```diff
@@ -410,6 +411,42 @@ func ValidateNodeCIDRMaskWithMaxPod(maxPod int32, nodeCIDRMaskSize int32) field.
 	return allErrs
 }
 
+// ValidateTotalNodeCountWithPodCIDR validates if the podCIDRs in the Pod Network can support the maximum number of nodes configured in the worker pools of the shoot
+func ValidateTotalNodeCountWithPodCIDR(shoot *core.Shoot) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	var (
+		totalNodes       int32
+		nodeCIDRMaskSize int32 = 24
+		podNetworkCIDR         = core.DefaultPodNetworkCIDR
+	)
+
+	if shoot.Spec.Networking.Pods != nil {
+		podNetworkCIDR = *shoot.Spec.Networking.Pods
+	}
+	if shoot.Spec.Kubernetes.KubeControllerManager != nil && shoot.Spec.Kubernetes.KubeControllerManager.NodeCIDRMaskSize != nil {
+		nodeCIDRMaskSize = *shoot.Spec.Kubernetes.KubeControllerManager.NodeCIDRMaskSize
+	}
+
+	_, podNetwork, err := net.ParseCIDR(podNetworkCIDR)
+	if err != nil {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("spec").Child("networking").Child("pods"), podNetworkCIDR, fmt.Sprintf("cannot parse shoot's pod network cidr : %s", podNetworkCIDR)))
+		return allErrs
+	}
+
+	cidrMask, _ := podNetwork.Mask.Size()
```

Comment:
Nit: `Mask.Size()` may return `(0,0)` if the mask is incorrect. Therefore, you could potentially check for this error condition. However, it is fair to assume that `net.ParseCIDR()` provides a proper mask.